Assets/MixedRealityToolkit.SDK/Features/Utilities/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,390 @@
+            }
+        }
+
+        /// <summary>
+        /// Gets the direction of the gaze for purposes of placing the stabilization plane
+        /// </summary>
+        private Vector3 GazeNormal
+        {
+            get
+            {
+                if (InputSystem.GazeProvider.Enabled)
+                {
+                    return InputSystem.GazeProvider.GazeDirection;
+                }
+                return Camera.main.transform.forward;
+            }
+        }
+
+        /// <summary>
+        /// Gets the position hit on the object the user is gazing at, if gaze tracking is supported.
+        /// </summary>
+        /// <param name="hitPosition">The position at which gaze ray intersects with an object.</param>
+        /// <returns>True if gaze is supported and an object was hit by gaze, otherwise false.</returns>
+        private bool TryGetGazeHitPosition(out Vector3 hitPosition)
+        {
+            if (InputSystem.GazeProvider.Enabled)
+            {
+                hitPosition = InputSystem.GazeProvider.HitPosition;
+                return true;
+            }
+            hitPosition = Vector3.zero;
+            return false;
+        }
+
+        /// <summary>
+        /// Configures the stabilization plane to update its position based on an object in the scene.        
+        /// </summary>
+        private void ConfigureTransformOverridePlane(float deltaTime)
+        {
+            planePosition = TargetOverride.position;
+
+            Vector3 gazeOrigin = GazeOrigin;
+            Vector3 gazeDirection = TargetOverride.position - InputSystem.GazeProvider.GazePointer.Position;
+            Vector3 velocity = Vector3.zero;
+            if (TrackVelocity)
+            {
+                velocity = UpdateVelocity(deltaTime);
+            }
+
+            float focusPointDistance = gazeDirection.magnitude;
+            float lerpPower = focusPointDistance > currentPlaneDistance ? LerpStabilizationPlanePowerFarther
+                                                                        : LerpStabilizationPlanePowerCloser;
+
+            // Smoothly move the focus point from previous hit position to new position.
+            currentPlaneDistance = Mathf.Lerp(currentPlaneDistance, focusPointDistance, lerpPower * deltaTime);
+
+            gazeDirection.Normalize();
+            planePosition = gazeOrigin + (gazeDirection * currentPlaneDistance);
+
+#if UNITY_EDITOR
+            debugPlane.Center = planePosition;
+            debugPlane.Normal = -gazeDirection;
+#elif UNITY_WSA
+            // Place the plane at the desired depth in front of the user and billboard it to the gaze origin.
+            HolographicSettings.SetFocusPointForFrame(planePosition, -gazeDirection, velocity);
+#endif
+        }
+
+        private void ConfigureOverridePlane(float deltaTime)
+        {
+            planePosition = OverridePlane.Center;
+
+            Vector3 velocity = Vector3.zero;
```

Comment:
nit: simpler velocity = TrackVelocity ? UpdateVelocity(deltaTime) : Vector3.zero;?

Reply:
I think the code here is more readable